### ISSUE TITLE
Type Any cannot be used in type signature

### ISSUE THREAD

**Issue by mitar** [python/mypy#4925]
I have a function which is returning dynamic enums. I am currently using `Any` as its return type:

```python
def create_enum_from_json_schema_enum(class_name: str, obj: typing.Dict, json_path: str, *, module: str = None, qualname: str = None, base_class: type = None) -> typing.Any:
    if qualname is None:
        qualname = class_name

    json_path_expression = jsonpath_ng.parse(json_path)

    names = [match.value for match in json_path_expression.find(obj)]

    # Make the list contain unique names. It works in Python 3.6+ because dicts are ordered.
    names = list(dict.fromkeys(names))

    return Enum(value=class_name, names=names, start=1, module=module, qualname=qualname, type=base_class)  # type: ignore

TaskType = utils.create_enum_from_json_schema_enum(
    'TaskType', base.DEFINITIONS_JSON,
    'definitions.problem.properties.task_type.oneOf[*].enum[*]',
    module=__name__,
)
```

The issue is that this works for accessing attributes of returned Enum class, `TaskType` example above. Works in the sense that it does not complain about it. But I cannot use `TaskType` in a type signature of any other function. Maybe something like `AnyType` is necessary?

I get `Invalid type "...TaskType"` error.

**Comment by ilevkivskyi** [comment]
No, just use `if TYPE_CHECKING: TaskType = Any ...`

**Comment by mitar** [comment]
But `TaskType` is already `Any`, no?

**Comment by JelleZijlstra** [comment]
It's of type Any, which is different from being a type alias for Any.

**Comment by ilevkivskyi** [comment]
It is a variable, not a type now.

**Comment by ilevkivskyi** [comment]
@mitar do you se the difference between
```
x: T
```
and 
```
x = T
```
? This is exactly the same case but where `T` is `Any`

**Comment by mitar** [comment]
I tried this, it does not work. So I have now:

```
if typing.TYPE_CHECKING:
    TaskType = typing.Any
else:
    TaskType = utils.create_enum_from_json_schema_enum(
        'TaskType', base.DEFINITIONS_JSON,
        'definitions.problem.properties.task_type.oneOf[*].enum[*]',
        module=__name__,
    )
```

But it fails on things like (getting an enum value from string):

```python
task_type = TaskType[task_description['type']]
```
```
Bad number of arguments for type alias, expected: 0, given: 1
Invalid type "task_description"
```

And I have multiple such accesses inside the same function, so I also get:

```
Cannot assign multiple types to name "task_type" without an explicit "Type[...]" annotation
``

Moreover, where I am trying to accessing attributes, I get:

```
"object" has no attribute "ENUM_FIELD"
```

Not sure why `Any` is seen as object?

**Comment by ilevkivskyi** [comment]
Use `# type: ignore` then. You need to choose between some runtime black magic and static typing.

**Comment by mitar** [comment]
I mean, I want to use static typing everywhere else. But this requires then to use `# type: ignore` for the whole signature of a function, no? Or everywhere I access it. This is too broad.

**Comment by JukkaL** [comment]
`# type: ignore` silences generated errors on that particular line. If the error is about a bad type, then that type gets replaced internally with `Any` but other types on the line will be fine.
